A30-xds-v3.md

```diff
@@ -0,0 +1,255 @@
+order to continue using the latest release of gRPC.  Older releases of
+gRPC will of course continue to function using the older versions of the
+xDS API.
+
+### Deciding What Version to Use
+
+Any given gRPC release will support two versions of xDS, versions N-1 and N.
+It will assume that all servers speak version N-1, and there will be a
+server feature defined in the bootstrap file (see below) to tell it that
+the server supports version N.  When speaking to the server, gRPC will use
+the highest version that is common to both itself and the server.
+
+For example, let's say that the gRPC binary supports versions N-1 and N.
+If there is no server feature in the bootstrap file telling it that the
+server supports version N, it will assume that the server supports only
+version N-1, and it will speak version N-1 to the server.
+
+However, if the bootstrap file does contain a server feature telling gRPC
+that the server supports version N, then it will believe that the server
+supports both versions N-1 and N, and it will speak version N to the server.
+
+### Server Features in the Bootstrap File
+
+The bootstrap file that configures the XdsClient code in gRPC will have
+a new `server_features` field added to it.  The new format will be as
+follows:
+
+```
+{
+  // The xDS server to talk to.  The value is an array to allow for a
+  // future change to add support for failing over to a secondary xDS server
+  // if the primary is down, but for now, only the first entry in the
+  // array will be used.
+  "xds_servers": [
+    {
+      "server_uri": <string containing URI of xds server>,
+      // List of channel creds; client will stop at the first type it
+      // supports.  This field is optional; if not specified, xDS will use
+      // the same channel creds as the backends, but with any associated
+      // call creds stripped off.
+      "channel_creds": [
+        {
+          "type": <string containing channel cred type>,
+          // The "config" field is optional; it may be missing if the
+          // credential type does not require config parameters.
+          "config": <JSON object containing config for the type>
+        }
+      ],
+
+      // THIS FIELD IS NEW!
+      // A list of features supported by the server.  New values will
+      // be added over time.  For forward compatibility reasons, the
+      // client will ignore any entry in the list that it does not
+      // understand, regardless of type.
+      "server_features": [ ... ]
+
+    }
+  ],
+  "node": <JSON form of Node proto>
+}
+```
+
+For each new version of xDS, a server feature will be defined.  The
+presence of that feature in the `server_features` list will indicate
+that the server supports that version of xDS.
+
+Note that because each release of gRPC will support only two versions of
+xDS and will always assume that servers support the lower of the two
+versions, a server feature added to support version N+1 will no longer
```

Comment:
"will no longer be used" can easily be misinterpreted. It would be used, but by older clients, and the bootstrap generator doesn't necessarily know the client version. This paragraph seems to muddy the waters instead of making things clearer.

Reply:
I've clarified the wording here.